### PR TITLE
Add Base Scope Guidelines

### PR DESCRIPTION
**Pull request by FichteFoll**
This document will be used as an experiment to gather experiments with
the suggested workflow and structure.

~~Many~~Some TODO comments are still left in, but I want to gather some feedback as soon as possible ~~while I sleep~~. Please do comment on questions I added to TODO comments and of course on anything else.

Some fundamental questions off the top of my head:

1. Do the structure and the headings I chose make sense? All level 2 headings should go into the template and be mandatory for every document.
1. Are there any factual errors?
1. ~~Do you think the contents of this general guideline fit into the same document or should embedding be split into a different document and this one renamed to "base scope"?~~

I will add a glossary for the terms I use later, but I'm not sure how to go about that yet. It will clarify on terms such as "scope name", "scope stack", and "sub-scope".

For discussion of the acceptance process I added #2, which I will fill out with some concrete proposals based on what I wrote in https://github.com/sublimehq/Packages/issues/1440 later.

---

[Rendered version](https://github.com/SublimeText/ScopeNamingGuidelines/blob/pr/general/documents/000-base-scope.md)

### CODE REVIEW

**Comment by FichteFoll** [comment]
As explained in the commit message of https://github.com/SublimeText/ScopeNamingGuidelines/pull/3/commits/02d6981620ca8ff166821815b6355284278cc8d6, I changed the scope of this SNG to concern itself with just the base scope, while also clearing out the unresolved questions section because there aren't any within its scope.

I'm undecided whether the general info stuff should be a guideline in itself, but I believe it should be and that it should receive the first or zeroth number. Either way, it should be done in a separate PR and not as the first PR, since I want to test the process here for normal-ish guidelines and not for the special general one.

File: documents/000-base-scope.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+- **C** uses a base scope of `source.c`.
+  For certain iterations of the language, 
+  the scope can be further specialized using `source.c.99`.
+
+- **C++** is a different language from C and uses `source.c++`.
+
+---
+
+- **Php** files are essentially HTML files
```
Comment (AmjadHD) [inline documents/000-base-scope.md]: Php -> PHP

**Comment by FichteFoll** [comment]
Final review phase is open. I don't know when I will merge this if there is no response.

File: documents/000-base-scope.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  is not spell-checked and auto completions are enabled.
+
+- `embedding` special use case for syntaxes
+   that are embedded later into the file.
+   The effect of an `embedding` scope is mostly transparent
+   and the text vs. source decision
+   is deferred to the following scope name.
+
+The second sub-scope level of the scope name
+should be based on the
```
Comment (keith-hall) [inline documents/000-base-scope.md]: Is something missing here? "should be based on the"... what? :)

File: documents/000-base-scope.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  Text in such syntaxes is spell-checked
+  and auto completion is disabled while typing.
+
+- `source` shall be used
+  for programming languages containing
+  instructions for a compiler, an interpreter, or similar.
+  Text in such syntaxes usually has special meaning,
+  is not spell-checked and auto completions are enabled.
+
+- `embedding` special use case for syntaxes
```
Comment (keith-hall) [inline documents/000-base-scope.md]: The other two points read more like sentences, but this one doesn't. Maybe: "embedding" *is a* special use case...

File: documents/000-base-scope.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+it may opt to use the `basic` sub-scope
+instead of simply omitting it.
+
+
+### Scope Suffix
+
+A scope suffix is used for individual tokens.
+The suffix is appended to every scope name
+that is assigned to a token,
+but there may only be one such suffix.
```
Comment (keith-hall) [inline documents/000-base-scope.md]: To be clear, this is not referring to uniqueness but that the suffix generally consists of one scope atom (using terminology from syntect here) - I misinterpreted at first until I read further down and thought there was a contradiction before realising.

File: documents/000-base-scope.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+---
+
+- **Php** files are essentially HTML files
+  with template sequences for inline code execution.
+  Because of this,
+  `embedding.php` is used as the base scope
+  and `text.html.basic` is the next scope name
+  immediately pushed onto the stack.
+  Embedded Php segments,
```
Comment (keith-hall) [inline documents/000-base-scope.md]: PHP is normally all caps, unless I'm mistaken?

File: documents/000-base-scope.md (hunk 2 above)
Comment (FichteFoll) [reply documents/000-base-scope.md]: Ah yes, good catch. Seems I forgot to finish that sentence or remove it as I was working on an alternative.

File: documents/000-base-scope.md (hunk 4 above)
Comment (FichteFoll) [reply documents/000-base-scope.md]: Good point, I'll rephrase that.

**Review by keith-hall — APPROVED** [review]

**Pull request merged**